flake.nix

```diff
@@ -2,48 +2,40 @@
   description = "hyperswitch";
 
   inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+
+    # TODO: Move away from these to https://github.com/juspay/rust-flake
     cargo2nix.url = "github:cargo2nix/cargo2nix/release-0.11.0";
     rust-overlay.url = "github:oxalica/rust-overlay";
-    flake-parts.url = "github:hercules-ci/flake-parts";
-    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
   };
 
-  outputs = inputs@{ self, nixpkgs, flake-parts, ... }:
-    flake-parts.lib.mkFlake { inherit inputs; } {
-      systems = nixpkgs.lib.systems.flakeExposed;
+  outputs = inputs:
+    inputs.flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = inputs.nixpkgs.lib.systems.flakeExposed;
       perSystem = { self', pkgs, system, ... }:
         let
           rustVersion = "1.65.0";
```

Comment:
Can we either use latest stable toolchain, or can we pick the MSRV from the `rust-version` field in top-level `Cargo.toml` file?

Reply:
Done.